### PR TITLE
Feature conect redux

### PR DESCRIPTION
**Pull request by dgsierra**
Changes:

- Connect React and Redux
- Add functional adding books
- Add functional removing books
- Add dependencies uuid, immer

Note: immer dependency is for helping with the immutability of the objects in src/redux/books/books.js

### CODE REVIEW

File: src/redux/categories/categories.js [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
+
+export const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    setCategorie: (state) => {
+      state.categories = 'COMMING SOON';
+    },
+  },
+});
```
Comment (codepantha) [inline src/redux/categories/categories.js]: - [ ] Please create a button for checking status from the `<Categories />` component. This button would use a function checkCategoriesStatus (name can be different) made in the component to dispatch the corresponding action creator 👍

File: src/components/BooksForm.js [hunk 2]
```diff
@@ -1,11 +1,29 @@
...
+import { setBook } from '../redux/books/books';
 
 export default function BookForm() {
+  const dispatch = useDispatch();
+  function submitHandler(e) {
+    e.preventDefault();
+    dispatch(setBook(
+      {
+        gender: e.target.children[1].value,
+        author: 'New Author',
```
Comment (codepantha) [inline src/components/BooksForm.js:12]: - [ ] According to requirements, please create a local state with `title` and `author` and they should have their inputs that update the local state every time they update (`onChange`)

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRES CHANGES ♻️ 

## Hi @dgonzalesi 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit messages
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BooksForm.js (hunk 2 above)
Comment (dgsierra) [reply src/components/BooksForm.js:12]: Its need to be in every component? As the local state is handle in the Books.js with setState in an array of objects, the BooksForm is passing the data to redux for make it global and stored in the App trough Books.js

Comment (dgsierra) [reply src/components/BooksForm.js:12]: @codepantha as Booksform the only function is get data and not display it thats way is only passing data to Redux, and Books.js handle the storage of the data.

File: src/redux/categories/categories.js (hunk 1 above)
Comment (rloterh) [reply src/redux/categories/categories.js]: Well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Daniel,  

Your project is complete! There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉🌟 

## Highlights 👍
- React successfully connected to redux

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**You can also consider:**
- MIT license link in your ReadMe is broken, kindly endeavor to fix it.
Implementing the above change(s) will make your ReadMe appear more professional.
<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BooksForm.js (hunk 2 above)
Comment (codepantha) [reply src/components/BooksForm.js:12]: @dgonzalesi  What you did works quite alright but the requirement was to make sure there was a local state

**Pull request merged**